File: AkitaModel/CardPayment.cs
namespace Skyware.Lis.AkitaModel;

public class CardPayment : PaymentBase
{

    /// <inheritdoc />
    public override PaymentMethod Method => PaymentMethod.Card;

    /// <summary>
    /// Card number (masked) used for the payment.
    /// </summary>
    public string CardNumber { get; set; }

}

File: AkitaMotelTS/src/CardPayment.ts
/**
 * Skyware.Lis.AkitaModel
 */

import { PaymentBase } from './PaymentBase';
import { PaymentMethod } from './PaymentMethod';

export class CardPayment extends PaymentBase {
  /** @inheritdoc */
  get method(): PaymentMethod {
    return PaymentMethod.Card;
  }

  /**
   * Card number (masked) used for the payment.
   */
  cardNumber?: string;
}
